backend/controller/SequenceController.ts

```diff
@@ -0,0 +1,384 @@
+        }
+
+        if (sequence.slides !== undefined) {
+            seq.slideCount = sequence.slides.length;
+            this.saveSlides(sequence.slides);
+        }
+
+        seq.save();
+    }
+
+    /**
+     * Deletes the sequence with the given code
+     * @param code the code of the sequence
+     */
+    public static async deleteSequence(code: string): Promise<void> {
+        const seq = await DBSequence.findOneBy({ code: code });
+        if (seq === null) {
+            throw new LoernwerkError(
+                'Sequence not Found',
+                LoernwerkErrorCodes.NOT_FOUND
+            );
+        }
+        const slides = await DBSlide.findBy({ sequenceCode: code });
+        for (const x of slides) {
+            x.remove();
+        }
+        // TODO: Remove H5P Content
+        for (const x of seq.readAccess) {
+            const user = await DBUser.findOneBy({ id: x });
+            if (user == null) {
+                continue;
+            }
+            user.sharedSequencesReadAccess = this.removeFromUserList(
+                user.sharedSequencesReadAccess,
+                code
+            );
+            user.save();
+        }
+        for (const x of seq.writeAccess) {
+            const user = await DBUser.findOneBy({ id: x });
+            if (user == null) {
+                continue;
+            }
+            user.sharedSequencesWriteAccess = this.removeFromUserList(
+                user.sharedSequencesWriteAccess,
+                code
+            );
+            user.save();
+        }
+        seq.remove();
+    }
+
+    /**
+     * Searches all sequenes of the user
+     * @param userId the id of the user
+     * @returns the sequences of the user
+     */
+    public static async getSequencesOfUser(
+        userId: number
+    ): Promise<ISequence[]> {
+        if (!(await this.isValidUser(userId))) {
+            throw new LoernwerkError(
+                'User doesnt exists',
+                LoernwerkErrorCodes.NOT_FOUND
+            );
+        }
+        return await DBSequence.find({ where: { authorId: userId } });
+    }
+
+    /**
+     * Searches all sequences shared with the user.
+     * @param userId the id of the user
+     * @returns the sequences shared with the user
+     */
+    public static async getSharedSequencesOfUser(
+        userId: number
+    ): Promise<ISequence[]> {
+        const user = await DBUser.findOneBy({ id: userId });
+        if (user === null) {
+            throw new LoernwerkError(
+                'User doesnt exists',
+                LoernwerkErrorCodes.NOT_FOUND
+            );
+        }
+        const seqlist: ISequence[] = [];
+        for (const x of user.sharedSequencesReadAccess.concat(
+            user.sharedSequencesWriteAccess
+        )) {
+            const seq = await DBSequence.findOneBy({ code: x });
+            if (seq === null) {
+                // May just bring it back to consistency?
+                throw new LoernwerkError(
```

Comment:
Should we throw an error, or just log it to console, but keep going?

Reply:
Could be a problem if we want to add someone to an access and it fails, then an error would be good.

Reply:
fair point, shouldn't happen either way